src/main/java/com/mucommander/main/muCommander.java

```diff
@@ -291,18 +291,25 @@ public static void main(String[] args) throws Exception
         muCommander.copySystemProperties(configProps);
 
         File preferencesFolder;
-        if (configuration.preferences == null) {
+        if (configuration.preferences != null) {
             try {
-                preferencesFolder = UserPreferencesDir.getDefaultPreferencesFolder();
+                preferencesFolder = UserPreferencesDir.getPreferencesFolder(configuration.preferences);
+            } catch(RuntimeException e) {
+                System.err.println("Failed to retrieve specified preferences folder: " + configuration.preferences);
+                return;
+            }
+        } else if (new File(codeParentFolder, ".portable").exists()) {

```

Comment:
Any reason for not creating the `.mucommander` folder within the installation directory instead of having the `.portable` file?

Reply:
I prefer having a clear distinction (the .portable file) of when to switch to portable mode, with the thinking that there may be more than one things being changed in the behaviour of the application (although in this case and for now there is only one thing being changed) and the .portable file is a kind of a flag for switching all of them. I also like it better being able to switch mode by removing/creating the .portable file instead of renaming the .mucommander folder. This is a personal preference though and I have seen it done both ways. If you think it would be better to have the .mucommander folder included in the distribution instead let me know and I will update the PR.

Reply:
I thought it would be better to create the `.mucommamnder` folder with the right permissions so different users would be able to execute the same portable app (imagine the application is first started by root so the `.mucommander` is set with permissions for the root user and so we won't be able to start the app later on by a non-root user). But even if the `.mucommander` folder is set with `777`, the files inside this folder would be created by the user that was the first to initiate the application so we'll have the same problem, I think...

Reply:
So maybe we should "remember" that we run in a portable mode and set the permission on the preferences file accordingly? what do you think?

Reply:
That makes sence and it would be very useful for your described scenario. If we proceed with that, I agree that remembering that we run in portable mode is how I feel it would be best to implement it as well. I am not sure if I like it as the default behaviour though. If I am not mistaken, in a multiuser environment with a user having muCommander in their home directory, this would allow other users to run it (I believe read permissions are usually allowed by default) and overwrite their settings, right? I think as a user I would not expect this to be allowed by default and although it would take a malicious user to cause issues I would still like to safeguard against it. In a situation where a user does want this to be allowed, they could manually set the permissions of the .mucommander folder to 777.

As a side note regarding .mucommander vs .portable, another scenario just crossed my mind. To update a portable installation, a user would probably copy the contents of the new version over the contents of the older installation. If they use macOS and Finder to do that, and chooses to replace/merge all, they would lose all their settings as the empty .mucommander folder from the new version would replace the old one which contained the user settings. Then again this could be considered a too specific scenario and a user error (and a bad default by Finder) but I also have fallen to this trap before myself and replaced the contents of folders I didn't mean to.

In any case, I do not feel strongly about any of the above and I am not certain about some of the assumptions I made. What do you think, are these valid concerns or are they too specific scenarios?

Reply:
Yeah, those are valid arguments/concerns.
I think it really depends on the use case.
Generally, if I copy a file to, let's say a USB drive, and plug it elsewhere - I'll have access to the copied file also as a different user. I think it was that way also when mucommander stored the configuration files within the app folder for portable versions so let's go with that.